packages/workflow-angular/lib/workflow-process/workflow-process-steps/workflow-process-steps-controller.js

```diff
@@ -46,18 +45,18 @@ function WorkflowProcessStepsController($scope, $state, mediator, workflowMediat
   }
 
   self.back = function() {
-    workflowMediatorService.previousStep(workorderId).then(function(workflowSummary) {
+    workflowApiService.previousStep(workorderId).then(function(workflowSummary) {
       updateWorkflowState(workflowSummary);
     });
   };
 
   //Beginning the workflow
-  workflowMediatorService.beginWorkflow(workorderId).then(function(workflowSummary) {
+  workflowApiService.beginWorkflow(workorderId).then(function(workflowSummary) {
     updateWorkflowState(workflowSummary);
   });
 
-  mediator.subscribeForScope("wfm:workflow:step:done", $scope, function(submission) {
-    workflowMediatorService.completeStep({
+  workflowApiService.nextStepSubscriber(function(submission) {
```

Comment:
That's going to be dropped anyway. They are temporary here until I will setup some base. 

> I know you don't like EventEmmiter

I love events. I like to use best technologies to do the job. 
I just do not like to use events without any reasons where there is easy way to do it without them. In this particular case we will need to use them for sure.